src/main/java/com/mesosphere/velocity/marathon/interfaces/AppConfig.java

```diff
@@ -21,11 +22,18 @@
     String getUrl();
 
     /**
+     * Get the configured docker map.
+     *
+     * @return Docker image name
+     */
+    Map<String, Object> getDocker();
```

Comment:
This should be the [Docker type](https://github.com/mesosphere/marathon-client/blob/master/src/main/java/mesosphere/marathon/client/model/v2/Docker.java) from [marathon-client](https://github.com/mesosphere/marathon-client/). It supports image name, force pull, and several other attributes. This will serialize to the config file correctly and prevent type casting further in the code.